src/query/all_query.rs

```diff
@@ -53,6 +54,30 @@ impl DocSet for AllScorer {
         self.doc
     }
 
+    fn fill_buffer(&mut self, buffer: &mut [DocId; BUFFER_LEN]) -> usize {
+        if self.doc() == TERMINATED {
+            return 0;
+        }
+        let is_safe_distance = self.doc() + (buffer.len() as u32) < self.max_doc;
```

Comment:
isn't that logic broken?
If a self.max_doc - self.doc() is not a good indication of the number of docs remaining in the docset.